### PR TITLE
Added function to compare chunks

### PR DESCRIPTION
**Pull request by ethaniccc**
This allows for two chunks with two different pointers to still be compared for equality. 
(pls merge so I don't have to update oomph's fork 🙏)

### CODE REVIEW

**Review by Sandertv — COMMENTED** [review]
Feel free to combine some of these checks into a single statement. That shouldn't lead to a longer execution time.

**Comment by ethaniccc** [comment]
👍

**Comment by ethaniccc** [comment]
bump

**Comment by RestartFU** [comment]
> bump

rebump

**Comment by cqdetdev** [comment]
bumpy

**Comment by RestartFU** [comment]
can we merge this? I'd like to be able to get rid of my dragonfly fork I'm currently using

**Review by RestartFU — CHANGES_REQUESTED** [review]
can you run `go fmt` :)

**Comment by didntpot** [comment]
@ethaniccc :V

File: server/world/chunk/sub_chunk.go [hunk 1]
```diff
@@ -14,6 +16,25 @@ func NewSubChunk(air uint32) *SubChunk {
 	return &SubChunk{air: air}
 }
 
+// Equals returns if the sub chunk passed is equal to the current one.
+func (sub *SubChunk) Equals(s *SubChunk) bool {
+	if s.air != sub.air || !slices.Equal(s.blockLight, sub.blockLight) ||
```
Comment (RestartFU) [inline server/world/chunk/sub_chunk.go:21]:
```suggestion
	if s.air != sub.air ||
```

File: server/world/chunk/sub_chunk.go [hunk 2]
```diff
@@ -14,6 +16,25 @@ func NewSubChunk(air uint32) *SubChunk {
...
+		!slices.Equal(s.skyLight, sub.skyLight) || len(s.storages) != len(sub.storages) {
```
Comment (RestartFU) [inline server/world/chunk/sub_chunk.go:22]:
```suggestion
		len(s.storages) != len(sub.storages) {
```

**Review by RestartFU — CHANGES_REQUESTED** [review]

**Pull request closed without merging**